kvbc/test/kvbc_app_filter/kvbc_filter_test.cpp

```diff
@@ -211,6 +284,34 @@ TEST(kvbc_filter_test, kvbfilter_update_success) {
   }
 }
 
+TEST(kvbc_filter_test, kvbfilter_update_success_eg) {
+  FakeStorage storage;
+  int client_id = 1;
+  const auto eg_id = EventGroupId{1};
+  auto kvb_filter = KvbAppFilter(&storage, std::to_string(client_id));
+
+  auto val_trid1 = CreateTridKvbValue("TridVal1", {"0", "1"});
+  auto val_trid2 = CreateTridKvbValue("TridVal2", {"0"});
+
+  concord::kvbc::categorization::EventGroup event_group{};
+  concord::kvbc::categorization::Event event1;
+  event1.data = val_trid1;
+  event1.tags = {"0", "1"};
+  concord::kvbc::categorization::Event event2;
+  event2.data = val_trid2;
+  event2.tags = {"0"};
+  event_group.events.emplace_back(event1);
+  event_group.events.emplace_back(event2);
+
+  const auto &filtered = kvb_filter.filterEventGroupUpdate({eg_id, event_group});
+
+  EXPECT_EQ(filtered.event_group_id, eg_id);
+  EXPECT_EQ(filtered.event_group.events.size(), 1);
+  for (auto &event : filtered.event_group.events) {
```

Comment:
Nit: We don't need a for-loop for a single event.

Reply:
Thanks for catching that. Will update in this PR if it fails (4th run 🤞 ), else will add the change in another PR.